### ISSUE TITLE
Restore a file from trashbin -> Undefined index: files at cloud/apps/files_trashbin/ajax/undelete.php#31

### ISSUE THREAD

**Issue by Dennis1993** [nextcloud/server#528]
### Steps to reproduce
1. delete a file
2. restore it from trashbin

### Actual behaviour
Tell us what happens instead
file is restored but error message in log

### Server configuration
**Operating system**:
Ubuntu

**Web server:**
Apache 2.4

**Database:**
MySQL 5.6.

**PHP version:**
7.0.8

**Nextcloud version:** (see Nextcloud admin page)
9.0.53 (stable)

**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh install

**Where did you install Nextcloud from:**
nextcloud.com

**List of activated apps:**
<details>
Default apps with calendar app
</details>

**The content of config/config.php:**
<details>
default
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
IE11, Microsoft Edge

**Operating system:**
Windows 10

#### Web server error log
<details>
<summary>Web server error log</summary>
```
Undefined index: files at cloud/apps/files_trashbin/ajax/undelete.php#31
```
</details>

**Comment by LukasReschke** [comment]
While I can see in the code what triggers this error message I can't quite understand when this code path is triggered.

Does this happen for any file? Or are they in subfolders or do you try to restore folders?

**Comment by Dennis1993** [comment]
I know what the problem.

I have deleted a file (picture) from the root. If I click now "Wiederherstellen" (restore file) in the same line, no error occurs. But if I select the file and use the button above the table -> error is logged.

Here a screenshot with a Little documentation what I've do.

Thanks for your reply!

![restore_error](https://cloud.githubusercontent.com/assets/1473674/17058345/367dc146-5020-11e6-834d-e812258718a7.PNG)

**Comment by MorrisJobke** [comment]
Fix is in #547
